script/shaders.ps1

```diff
@@ -0,0 +1,89 @@
+param
+(
+[Parameter(Mandatory=$false)]
+[string]$InputDir = "" + $(get-location).Path + "\resources\shaders",
+
+[Parameter(Mandatory=$false)]
+[string]$OutputDir = "" + $(get-location).Path + "\resources\shaders\compiled"
+)
+
+$Compiler = $Env:VULKAN_SDK + "\Bin\glslangValidator.exe"
+$Optimizer = $Env:VULKAN_SDK + "\Bin\spirv-opt.exe"
+$Preamble = "#pragma pack_matrix(row_major)"
+$TempDir = Join-Path $OutputDir "temp"
+
+$FailureCount = 0
+
+if (-not (Test-Path -Path $TempDir)) {
+    New-Item -Path $TempDir -ItemType Directory | Out-Null
+}
+
+Write-Host "--Compiling Shaders--"
+Get-ChildItem -Path $InputDir | Where-Object { $_.Extension -eq '.psh' -or $_.Extension -eq '.vsh' } |
+ForEach-Object {
+    if ($_.Extension -eq '.psh') {
+        $Stage = 'frag'
+        $FileName = "$($_.BaseName)Fragment.spv"
+    } elseif ($_.Extension -eq '.vsh') {
+        $Stage = 'vert'
+        $FileName = "$($_.BaseName)Vertex.spv"
+    } else {
+        return
+    }
+
+    Write-Output "Compiling $_"
+    $InputFile = Join-Path $InputDir $_.Name
+    $OutputFile = Join-Path $TempDir $FileName
+    $Args = @(
+        '-V',
+        '-D',
+        "-P$Preamble",
+        '-e', 'main',
+        '-fhlsl_functionality1',
+        "-S", $Stage,
+        "-o", $OutputFile,
+        $InputFile
+    )
+
+    & $Compiler @Args
```

Comment:
Can you add "-g" (compile in debug mode) to the possible parameters? I haven't been able to figure out how to do this since switching to Diligent and it would make a HUGE difference. I'm assuming we also don't want to optimize if running in this mode.

Debug allows you to see the actual shader source code when stepping through in Render Doc, and its hugely valuable

Reply:
Lets try to go over this today. Debug symbols exist, but spirv-opt may be generating a wildly different result. Not sure how that will affect your workflow. If we want to have two distinct build options, I'd prefer doing something like a debug/release directory that we can select from via config. Because the binaries are checked into source control, I don't want to leave open the possibility of accidentally checking in debug versions. I suppose we don't have to check in the binaries at all. Idk, worth a discussion.